frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -18,25 +23,50 @@ export const ModalContext = React.createContext<ModalContextValue>({
   closeModal: () => {},
 });
 
+type ComponentMap = {
+  [key: string]: {
+    Component: ModalComponent<UnknownProps>;
```

Comment:
Nit - type parameter `P` of `ModalComponent<P>` has a fallback to `UnknownProps` so we can omit it here:
```suggestion
    Component: ModalComponent;
```